src/webots/gui/WbX3dStreamingServer.cpp

```diff
@@ -65,9 +65,11 @@ void WbX3dStreamingServer::create(int port) {
 
 void WbX3dStreamingServer::sendTcpRequestReply(const QString &url, const QString &etag, const QString &host,
                                                QTcpSocket *socket) {
-  QFileInfo file(WbProject::current()->dir().absolutePath() + "/" + url);
+  QString urlWithSpaces(url);
+  urlWithSpaces.replace("%20", " ");
```

Comment:
`%20` is only an example of percent encoding, but there are many more. To be generic, we should probably use:
```c++
const QString decodedUrl = QUrl::fromPercentEncoding(url.toUtf8());
```